src/backend/executor/spi.c

```diff
@@ -1982,7 +1982,7 @@ _SPI_prepare_plan(const char *src, SPIPlanPtr plan)
 		CompleteCachedPlan(plansource,
 						   stmt_list,
 						   NULL,
-						   nodeTag(parsetree),
+						   parsetree ? nodeTag(parsetree) : T_Invalid,
```

Comment:
Your solution does not solve the `CreateCommandTag(parsetree)` problem above, while [mine](https://github.com/arenadata/gpdb/pull/732/files#diff-eca689b02874d585dbe0547b5b5af6e723a88247c89b662fa0f601f3d363d4f5R1950) does!